### PR TITLE
Fix LRO tests in the autorest.java repository

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: azure-client-runtime/src/main/java/com/microsoft/azure/AsyncOperationResource.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    public String status() {
+        return status;
+    }
+
+    /**
+     * Set the status of the asynchronous operation.
+     * @param status The status of the asynchronous operation.
+     */
+    public void status(String status) {
```
Comment (RikkiGibson) [inline azure-client-runtime/src/main/java/com/microsoft/azure/AsyncOperationResource.java]: `setStatus`?

File: azure-client-runtime/src/main/java/com/microsoft/azure/AzureAsyncOperationPollStrategy.java [hunk 2]
```diff
@@ -73,37 +75,50 @@ else if (pollingSucceeded) {
 
     @Override
     public Single<HttpResponse> updateFromAsync(final HttpResponse httpPollResponse) {
+        if (400 <= httpPollResponse.statusCode()) {
```
Comment (RikkiGibson) [inline azure-client-runtime/src/main/java/com/microsoft/azure/AzureAsyncOperationPollStrategy.java]: Not a big thing but I encourage putting constants on the right side of a comparison

File: azure-client-runtime/src/main/java/com/microsoft/azure/AzureProxy.java [hunk 3]
```diff
@@ -140,72 +141,108 @@ protected Object handleAsyncHttpResponse(final HttpRequest httpRequest, Single<H
...
+
+    private Single<PollStrategy> createProvisioningStateOrCompletedPollStrategy(final HttpRequest httpRequest, HttpResponse httpResponse, SwaggerMethodParser methodParser, final long delayInMilliseconds) {
+        Single<PollStrategy> result;
+
+        final String httpRequestMethod = httpRequest.httpMethod();
+        if (httpRequestMethod.equalsIgnoreCase("DELETE") ||
+                httpRequestMethod.equalsIgnoreCase("GET") ||
+                httpRequestMethod.equalsIgnoreCase("HEAD") ||
+                !methodParser.expectsResponseBody()) {
+            result = Single.<PollStrategy>just(new CompletedPollStrategy(AzureProxy.this, httpResponse));
```
Comment (RikkiGibson) [inline azure-client-runtime/src/main/java/com/microsoft/azure/AzureProxy.java]: consider checking content-type: application/json here

Comment (RikkiGibson) [reply azure-client-runtime/src/main/java/com/microsoft/azure/AzureProxy.java]: This way we won't have to buffer octet streams and attempt to parse them as JSON before we give them over

Comment (RikkiGibson) [reply azure-client-runtime/src/main/java/com/microsoft/azure/AzureProxy.java]: My only fear is that autorest.testserver isn't giving a response Content-Type back--but we'll be able to work around it, maybe even with a requestpolicy that adds it in the worst case

File: azure-client-runtime/src/main/java/com/microsoft/azure/LocationPollStrategy.java [hunk 4]
```diff
@@ -64,17 +69,41 @@ public boolean isDone() {
...
-    static PollStrategy tryToCreate(RestProxy restProxy, String fullyQualifiedMethodName, HttpResponse httpResponse, long delayInMilliseconds) {
+    static PollStrategy tryToCreate(RestProxy restProxy, HttpRequest originalHttpRequest, HttpResponse httpResponse, long delayInMilliseconds) {
         final String locationUrl = httpResponse.headerValue(HEADER_NAME);
-        return locationUrl != null && !locationUrl.isEmpty()
-                ? new LocationPollStrategy(restProxy, fullyQualifiedMethodName, locationUrl, delayInMilliseconds)
-                : null;
+
+        String pollUrl = null;
+        if (locationUrl != null && !locationUrl.isEmpty()) {
+            if (locationUrl.startsWith("/")) {
```
Comment (RikkiGibson) [inline azure-client-runtime/src/main/java/com/microsoft/azure/LocationPollStrategy.java]: can consider using https://stackoverflow.com/questions/8273019/how-to-merge-two-uris-in-java#8273086 to combine a base URL with a relative URL

File: azure-client-runtime/src/main/java/com/microsoft/azure/AsyncOperationResource.java (hunk 1 above)
Comment (daschult) [reply azure-client-runtime/src/main/java/com/microsoft/azure/AsyncOperationResource.java]: Done.

File: azure-client-runtime/src/main/java/com/microsoft/azure/AzureAsyncOperationPollStrategy.java (hunk 2 above)
Comment (daschult) [reply azure-client-runtime/src/main/java/com/microsoft/azure/AzureAsyncOperationPollStrategy.java]: Done.

File: azure-client-runtime/src/main/java/com/microsoft/azure/LocationPollStrategy.java (hunk 4 above)
Comment (daschult) [reply azure-client-runtime/src/main/java/com/microsoft/azure/LocationPollStrategy.java]: Done.

**Comment by daschult** [comment]
@RikkiGibson Thanks for fixing my checkstyle issues. :)

**Pull request merged**